features/netsocket/nsapi_dns.cpp

```diff
@@ -414,6 +423,7 @@ static void nsapi_dns_cache_reset()
             dns_cache[i]->host = NULL;
             delete dns_cache[i];
             dns_cache[i] = NULL;
+            delete dns_cache[i]->address;
```

Comment:
If cache entry is now allocated with `new[]`, then don't we also need to call `delete[]` to free it?